backend/api/Areas/Admin/Controllers/AgencyController.cs

```diff
@@ -98,11 +104,18 @@ public IActionResult GetAgencies(EModel.AgencyFilter filter)
         [ProducesResponseType(typeof(Model.AgencyModel), 201)]
         [ProducesResponseType(typeof(Api.Models.ErrorResponseModel), 400)]
         [SwaggerOperation(Tags = new[] { "admin-agency" })]
-        public IActionResult AddAgency([FromBody] Model.AgencyModel model)
+        public async Task<IActionResult> AddAgencyAsync([FromBody] Model.AgencyModel model)
         {
             var entity = _mapper.Map<Entity.Agency>(model);
             _pimsAdminService.Agency.Add(entity);
 
+            // TODO: This isn't ideal as the db update may be successful but this request may not.
+            await entity.Users.ForEachAsync(async u =>
```

Comment:
Need to tell Keycloak to update the users that were affected by the agency change.